identity/src/main/java/com/android/identity/DataTransportNfc.java

```diff
@@ -152,14 +152,21 @@ List<DataRetrievalAddress> parseDeviceRetrievalMethod(int version, @NonNull Data
         }
         Map options = ((Map) items[2]);
 
-        int commandDataFieldMaxLength = Util.cborMapExtractNumber(options,
+        long commandDataFieldMaxLength = Util.cborMapExtractNumber(options,
                 RETRIEVAL_OPTION_KEY_COMMAND_DATA_FIELD_MAX_LENGTH);
-        int responseDataFieldMaxLength = Util.cborMapExtractNumber(options,
+        long responseDataFieldMaxLength = Util.cborMapExtractNumber(options,
                 RETRIEVAL_OPTION_KEY_RESPONSE_DATA_FIELD_MAX_LENGTH);
 
+        if (commandDataFieldMaxLength > Integer.MAX_VALUE ||
+                responseDataFieldMaxLength  > Integer.MAX_VALUE) {
```

Comment:
I thought I already commented here but it seems to have gotten lost, or perhaps it was on another file?

I think we already need to check for < Integer.MIN_VALUE, no? I think the easiest way to check whether a long value fits into an int is probably to check whether

longValue == (int) longValue

for example via a helper method

Reply:
Sorry, I misread this comment originally. This should also check that the value is > 0.